pulp_container/app/serializers.py

```diff
@@ -288,6 +293,12 @@ def validate(self, data):
                 )
         return validated_data
 
+    def is_base_path_valid(self, base_path):
+        """Check whether the passed repository base path is valid or not."""
+        component_re = re.compile(r"^[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?$")
```

Comment:
I am a bit unsure, how long these processes live. Do you think it's worth storing the compiled regex in a global var?